#!/usr/bin/env python
PACKAGE = "volta_hardware"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Save_to_eeprom"		,   	bool_t		,	0	, "save to eeprom parameter"		,	False			)
gen.add("write_controls"		,   	bool_t		,	0	, "write pid parameters"		,	False			)
gen.add("reset_pid"			,   	bool_t		,	0	, "reset pid parameters"		,	False			)
gen.add("Kp1"		 		,    	double_t		,	0	, "motor one P-gain"		,	0.0	,	0.0,   100.0)
gen.add("Ki1"	 			,   	double_t		,	0	, "motor one I-gain"		,	0.0	,	0.0,   100.0)
gen.add("Kd1"		 		, 	double_t		,	0	, "motor one D-gain"		,	0.0	,	0.0,   100.0)
gen.add("Kp2"		 		,    	double_t		,	0	, "motor Two  P-gain"		,	0.0	,	0.0,   100.0)
gen.add("Ki2"	 			,   	double_t		,	0	, "motor Two  I-gain"		,	0.0	,	0.0,   100.0)
gen.add("Kd2"		 		, 	double_t		,	0	, "motor Two  D-gain"		,	0.0	,	0.0,   100.0)

exit(gen.generate(PACKAGE, "volta_hardware", "controls"))